### PR TITLE
EVC-14 pipeline fix rebased on boilerplate

### PR DESCRIPTION
**Pull request by michaelhartHO**
## What? 

Pipeline fixes rebased on the EVC-27-app-boilerplate branch
Deployment to branch 

# WHY ?
The files were updated and so that deployment to branch is successful

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -58,7 +58,7 @@ steps:
       SEVERITY: MEDIUM,HIGH,CRITICAL  --dependency-tree
       FAIL_ON_DETECTION: false
       IGNORE_UNFIXED: false
-      ALLOW_CVE_LIST_FILE: hof-services-config/infrastructure/trivy/.trivyignore.yaml
+      ALLOW_CVE_LIST_FILE: hof-services-config//E_Visa_Contact_Us/
```
Comment (adityababumallisettiHO) [inline .drone.yml:61]: @DivineTaminangHO we are using trivyignore as global cve file . 
The one you added is also incorrect it has two forward slashes

File: .drone.yml [hunk 2]
```diff
@@ -74,7 +74,7 @@ steps:
     image: 340268328991.dkr.ecr.eu-west-2.amazonaws.com/acp/dind
     environment:
       #DOCKER_PASSWORD:
-       # from_secret: DOCKER_PASSWORD
+       #from_secret: DOCKER_PASSWORD
```
Comment (adityababumallisettiHO) [inline .drone.yml:77]: @DivineTaminangHO Please delete anything in comments which is not being used by pipeline

File: kube/app/ingress-internal.yml [hunk 3]
```diff
@@ -13,8 +13,8 @@ spec:
   tls:
     - hosts:
       {{ if eq .KUBE_NAMESPACE .BRANCH_ENV }}
-        - {{ .DRONE_BUILD_NUMBER }}.internal.{{ .BRANCH_ENV }}.homeoffice.gov.uk
-        - {{ .APP_NAME }}-{{ .DRONE_SOURCE_BRANCH }}.internal.{{ .BRANCH_ENV }}.homeoffice.gov.uk
+        - {{ .DRONE_BUILD_NUMBER }}.internal.branch.sas-notprod.homeoffice.gov.uk
```
Comment (adityababumallisettiHO) [inline kube/app/ingress-internal.yml:16]: @DivineTaminangHO  We removed line 16 for COA which we feel as no use. We are only using internal ingress for each deployment which should have drone source branch.

**Review by adityababumallisettiHO — CHANGES_REQUESTED** [review]

File: apps/evisa/index.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    '/biometric-residence-permit-number': {
+      fields: [
+        'biometric-residence-permit-number-options',
+        'biometric-residence-permit-number-number'
+      ],
+      next: '/start1'
+    },
+
+    '/start1': {
+      next: '/start2',
```
Comment (william-gu-hof) [inline apps/evisa/index.js:26]: these step slugs are made up from when I wanted to code the project, as BA requirement gathering was not finished yet, these should be modified to fit with what's on the design, but it is okay as the start up repo of the project

File: apps/evisa/behaviours/agent-email.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+const hof = require('hof');
+const notify = hof.components.notify;
+// const { notify } = require('hof');   // hof.components.notify // TODO refactor to use this
+const path = require('path');
+
+const parse = (model, translate) => {
+  console.log('=====================');
+  console.log(model.images[0].url);
+  console.log('=====================');
```
Comment (william-gu-hof) [inline apps/evisa/behaviours/agent-email.js:12]: remember to take out the console logs :D

File: apps/evisa/behaviours/limit-documents.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+module.exports = superclass => class LimitDocs extends superclass {
+  validate(req, res, next) {
+    const images = req.sessionModel.get('images');
+    if (images && images.length >= 2) { // TODO replace magic number with config
```
Comment (william-gu-hof) [inline apps/evisa/behaviours/limit-documents.js:4]: I set the limit to 2 for testing, should be updated to what is required from BA

**Review by william-gu-hof — APPROVED** [review]
left some comments on few things that should be changed/updated, but should be okay as the starting repo

File: apps/evisa/behaviours/agent-email.js (hunk 5 above)
Comment (michaelhartHO) [reply apps/evisa/behaviours/agent-email.js:12]: Will do, they're coming in handy atm.

File: apps/evisa/behaviours/limit-documents.js (hunk 6 above)
Comment (michaelhartHO) [reply apps/evisa/behaviours/limit-documents.js:4]: If it's a hard-coded value I'll give it the `const` treatment

**Review by adityababumallisettiHO — APPROVED** [review]
lgtm

File: package.json [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/UKHomeOffice/evc/issues"
+  },
+  "homepage": "https://github.com/UKHomeOffice/evc#readme",
+  "dependencies": {
+    "accessible-autocomplete": "^2.0.4",
+    "bl": "^6.0.12",
+    "busboy": "^1.6.0",
+    "hof": "^20.5.6",
```
[the remark this answers is not in the document]
Comment (michaelhartHO) [reply package.json:31]: addressed upstream

File: config.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  env: env,
+  email: {
+    notifyApiKey: process.env.NOTIFY_KEY,
+    notifyTemplate: process.env.NOTIFY_TEMPLATE,
+    caseWorker: process.env.CASEWORKER_EMAIL
+  },
+  survey: {
+    urls: {
+      root: 'https://eforms.homeoffice.gov.uk/outreach/Feedback.ofml?FormName=coa/',
+      acq: 'https://eforms.homeoffice.gov.uk/outreach/Feedback.ofml?FormName=coa/'
```
[the remark this answers is not in the document]
Comment (michaelhartHO) [reply config.js:18]: addressed upstream

File: config.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  dateTimeFormat: 'DD MMM YYYY HH:mm:ss',
+  env: env,
+  email: {
+    notifyApiKey: process.env.NOTIFY_KEY,
+    notifyTemplate: process.env.NOTIFY_TEMPLATE,
+    caseWorker: process.env.CASEWORKER_EMAIL
+  },
+  survey: {
+    urls: {
+      root: 'https://eforms.homeoffice.gov.uk/outreach/Feedback.ofml?FormName=coa/',
```
[the remark this answers is not in the document]
Comment (michaelhartHO) [reply config.js:17]: addressed upstream

**Pull request merged**